src/MagicWormhole/Internal/FileTransfer.hs

```diff
@@ -29,12 +29,26 @@ import System.Posix.Types (FileOffset)
 data Offer
   -- | A simple text message.
   = Message Text
+  -- | Offer a File with filename and size.
   | File FilePath FileOffset
+  -- | Offer a Directory
+  | Directory
+    Text
+    -- ^ Mode. Currently always "zipfile/deflated".
```

Comment:
I don't know enough about this layer of Magic Wormhole, but is this really an arbitrary unicode string? Would this be better encoded as a mime type or enum or something?

Reply:
Yes, it is a hardcoded arbitrary unicode string. The python client checks if the sender sends this exact string.